test_user.go

```diff
@@ -37,6 +44,31 @@ func generateUserWallet(ctx context.Context, keyName string, chain ibc.Chain) (*
 	return &user, nil
 }
 
+// GetAndFundTestUserWithMnemonic restores a user using the given mnemonic
+// and funds it with the native chain denom.
+func GetAndFundTestUserWithMnemonic(
+	t *testing.T,
+	ctx context.Context,
+	keyNamePrefix, mnemonic string,
+	amount int64,
+	chain ibc.Chain,
+) *User {
+	chainCfg := chain.Config()
+	keyName := fmt.Sprintf("%s-%s-%s", keyNamePrefix, chainCfg.ChainID, dockerutil.RandLowerCaseLetterString(3))
+	user, err := generateUserWallet(ctx, keyName, mnemonic, chain)
+	require.NoError(t, err, "failed to get source user wallet")
+
+	err = chain.SendFunds(ctx, FaucetAccountKeyName, ibc.WalletAmount{
+		Address: user.Bech32Address(chainCfg.Bech32Prefix),
+		Amount:  amount,
+		Denom:   chainCfg.Denom,
+	})
+	require.NoError(t, err, "failed to get funds from faucet")
+	require.NoError(t, test.WaitForBlocks(ctx, 5, chain), "failed to wait for blocks")
```

Comment:
Did 5 come from anywhere special? Would 4 or 3 work?

Some of these tests can take a long time to complete, so if there is a way to minimize time spent waiting, we should take it.

Maybe it would make more sense to not wait inside this function, and indicate in Godoc that the caller should wait for some blocks to complete before the funds will be accessible. There is one wait at the end of `GetAndFundTestUsers` that would be redundant with this call, if it stays.

Reply:
This wait was just taken from the existing `GetAndFundTestUsers` function, however I think it's no harm removing it and letting the caller know as you suggested. I can remove from both.